code/modules/mob/living/simple_animal/hostile/grue.dm

```diff
@@ -0,0 +1,591 @@
+		moultcost=0 //not needed for adults
+		melee_damage_lower = 20
+		melee_damage_upper = 30
+		melee_damage_type = BRUTE
+		held_items = list()
+		burnmalus=3**(1/3)
+		environment_smash_flags = SMASH_LIGHT_STRUCTURES | SMASH_CONTAINERS | OPEN_DOOR_WEAK | OPEN_DOOR_STRONG
+		attack_sound = 'sound/weapons/cbar_hitbod1.ogg'
+		size = SIZE_BIG
+		pass_flags = 0
+		force_airlock_time=0
+	health=tempHealth*maxHealth
+
+//Grue vision
+/mob/living/simple_animal/hostile/grue/update_perception()
+
+	if(client)
+		if(client.darkness_planemaster&&a_use_alpha)
+			client.darkness_planemaster.blend_mode = BLEND_ADD
+			client.darkness_planemaster.alpha = a_blend_add_test
+		client.color = list(
+					1,0,0,1,
+					-1,0.2,0.2,1,
+	 				-1,0.2,0.2,1,
+		 			0,0,0,1,
+		 			0,0,0,0)
+
+		if(a_matrix_testing_override) //not used for now
+			client.color = list(a_11,a_12,a_13,a_14,
+								a_21,a_22,a_23,a_24,
+		 						a_31,a_32,a_33,a_34,
+			 					a_41,a_42,a_43,a_44,
+			 					a_51,a_52,a_53,a_54)
+//		else if(b_matrix_testing_override) //not used for now
+//			colourmatrix = list(b_11, b_12, b_22,
+//						 b_21,b_22, b_23,
+//						 b_31, b_32, b_33,
+//						 b_41, b_42, b_43)
+
+/mob/living/simple_animal/hostile/grue/Stat()
+	..()
+	if(statpanel("Status"))
+//		stat(null, "Intent: [a_intent]")
+//		stat(null, "Move Mode: [m_intent]")
+		if(ismoulting)
+			stat(null, "Moulting progress: [round(100*(1-moulttimer/moulttime),0.1)]%")
+		if(lifestage<3) //not needed for adults
+			stat(null, "Shadow power: [round(shadowpower,0.1)]/[round(maxshadowpower,0.1)]")
+		if(lifestage>=3)
+			if(config.grue_egglaying)
+				stat(null, "Reproductive energy: [eatencharge]")
+			stat(null, "Sentient life forms eaten: [eatencount]")
+
+/mob/living/simple_animal/hostile/grue/gruespawn
+	lifestage=1
+
+/mob/living/simple_animal/hostile/grue/grueling
+	lifestage=2
+
+//Moulting into more mature forms.
+/mob/living/simple_animal/hostile/grue/verb/moult()
+	set name = "Moult"
+	set desc = "Moult into a new form." //hide if an adult?
+	set category = "Grue"
+	if(!alert(src,"Would you like to moult? You will become a vulnerable and immobile chrysalis during the process.",,"Moult","Cancel") == "Moult")
+		return
+	if (lifestage<3)
+		if (shadowpower<moultcost)
+			to_chat(src, "<span class='notice'>You need to bask in shadow more first. ([moultcost] shadow power required)</span>")
+			return
+		else if (!isturf(loc))
+			to_chat(src, "<span class='notice'>You need more room to moult.</span>")
+			return
+		else if (stat==UNCONSCIOUS)
+			to_chat(src, "<span class='notice'>You must be awake to moult.</span>")
+			return
+		else if (busy)
+			to_chat(src, "<span class='notice'>You are already doing something.</span>")
+			return
+		else
+			start_moult()
+
+	else
+		to_chat(src, "<span class='notice'>You are already fully mature.</span>")
+
+/mob/living/simple_animal/hostile/grue/proc/start_moult()
+	if(stat==CONSCIOUS&&shadowpower>=moultcost&&!ismoulting&&lifestage<3)
```

Comment:
Style comment : space between operators (at leat for && and ==)

Reply:
Done.